src/mrpro/data/_kdata/KData.py

```diff
@@ -109,6 +109,40 @@ def from_file(
             modification_time = datetime.datetime.fromtimestamp(mtime)
 
         acquisitions = [acq for acq in acquisitions if acquisition_filter_criterion(acq)]
+
+        # Select only acquisitions with the desired number of receiver channels/coils.
+        # If the number of coils is not defined use all acquisitions
```

Comment:
i think think comment is wrong
# Ensure consistent number of receiver coils by filtering, default to the number matching the header

and move the` # We need to handle the n_coils parameter of header_overwrites here separately rather than` ...
into the if statement where you use use it.

and add a `best-effort guess` or something like that to the max(n_coils_available) line?